--- usr/include/types.h.b	2014-10-26 23:17:10.000000000 +0400
+++ usr/include/types.h	2017-06-10 21:28:04.000000000 +0400
@@ -23,7 +23,9 @@
 
 #define NIL ((char *) 0)
 #define PZERO 0
+#if 0 /* Don't pretend we're BSD (see sys/param.h) */
 #define BSD 43
+#endif
 #ifndef TRUE
 #define TRUE 1
 #endif
--- usr/include/sys/param.h.b	2014-10-26 23:17:10.000000000 +0400
+++ usr/include/sys/param.h	2017-06-10 21:26:18.000000000 +0400
@@ -23,6 +23,7 @@
 #endif
 
 
+#if 0 /* Don't pretend we're BSD (see below) */
 /** @group BSD version defines.
  * OS2: The toolkit headers define these. Resent FreeBSD release does too.
  * Warning! Be aware that config scripts and programs may check for these and
@@ -43,6 +44,7 @@
 #define BSD4_4  1
 #endif
 /** @} */
+#endif
 
 
 #ifndef LOCORE
