Index: emxomf.c
===================================================================
--- emxomf.c	(revision 3859)
+++ emxomf.c	(working copy)
@@ -2726,7 +2726,24 @@
                         const char *pszExpName, size_t cchExpName, unsigned iOrdinal)
 {
     const struct nlist *pSym;
-    char *pszSym = alloca(cchSymbol + 1);
+    char *pszSym;
+
+    /*
+     * GCC 4.4.2 prepends an asterisk to exported symbols having the _System
+     * attribute, remove it
+     */
+    if (*pszSymbol == '*')
+    {
+        ++pszSymbol;
+        --cchSymbol;
+    }
+    if (*pszExpName == '*')
+    {
+        ++pszExpName;
+        --cchExpName;
+    }
+
+    pszSym = alloca(cchSymbol + 1);
     memcpy(pszSym, pszSymbol, cchSymbol);
     pszSym[cchSymbol] = '\0';
 
